packages/core/src/utils/transformers/transformBoundContentValue.ts

```diff
@@ -24,6 +24,10 @@ export const transformBoundContentValue = (
 
   switch (variableDefinition.type) {
     case 'Media':
+      // If we bound a normal entry field to the media veriable we just return the bound value
+      if (entityOrAsset.sys.type === 'Entry') {
+        return getBoundValue(entityOrAsset, path);
+      }
```

Comment:
I think that this if-guard is really elegant in solving the problem: 
- it still allows to `Media>Txt` fields when binding to normal entries
- it guards `transformMedia(asset)` from receiving entiTies as an argument (note: before it was possible due to `entryOrAsset as Asset` cast; and this was handled by tailing `return getBoundValue(asset, path);` call within `transformMedia()` . And that wasn't very semantic